src/sync/tracks.nim

```diff
@@ -40,8 +41,10 @@ proc getPracticeExerciseSlugs(trackDir: TrackDir): seq[PracticeExerciseSlug] =
         result = newSeqOfCap[PracticeExerciseSlug](practiceExercises.len)
 
         for exercise in practiceExercises:
-          if exercise.hasKey("slug"):
-            if exercise["slug"].kind == JString:
+          if exercise.hasKey("slug") and exercise["slug"].kind == JString:
+            if withDeprecated or not exercise.hasKey("status") or
+                exercise["status"].kind != JString or
+                exercise["status"].getStr() != "deprecated":
```

Comment:
> for a long time now I've wanted a rewrite to use jsony everywhere - it would make the codebase much easier to read, maintain, and test.

Absolutely!